aces/Loris
# [redcap] new module

## Brief summary of changes

This PR adds the REDCap interoperability module.
This module opens an API endpoint to receive notifications from REDCap servers through the use of the `Data Entry Trigger (DET)`. REDCap notifications are parsed and REDCap records are imported into LORIS.
If an error happens during this notification handling process, an error is generated in the issue tracker.
The module is imported from HBCD, and even if the core logic is the same, REDCap objects are now their own models.
This version also supports connection to multiple REDCap instances.

### Features

The main goal was to replicate HBCD data import process based on `REDCap DET`.

- [x] Refactoring of the REDCap HTTP Client and linked classes/models.
- [x] Support for multiple REDCap instances.
- [x] Additional tools and features:
  - [x] `redcap2linst`: script to import REDCap forms as LORIS LINST instruments.
  - ~~[ ] `fetch button`: UI button on instrument panel to trigger data import~~ will be done in a later PR

### State

- [x] core module
  - [x] redcap2linst adapted from HBCD usage (originally from #8181). Added possibility to generate LINST instruments from different REDCap instances and files.
  - [ ] automated test
  - [x] test plan
- [x] RB configuration

#### Testing instructions (if applicable)

Notification process:
1. Define main assignee, REDCap instances and projects in the `config.xml` following the structure given in `RB/config.xml` file.
2. Define which instruments could be imported by adding them to the configuration list in LORIS. See `Front-end > Admin > Configuration > REDCap > redcap_importable_instrument`.
3. Define which user is the main issue tracker assignee for all redcap issues in the configuration. See `Front-end > Admin > Configuration > REDCap > redcap_issue_assignee`.
4. Send a URL notification to LORIS (i.e. endpoint `/redcap/notifications`) to simulate the reception of a DET notification. The payload should be the following (see `redcap/notifications/redcapnotification.class.inc`):
```json
{
    "instrument":"instrument_backend_name",
    "project_id":"REDCap project ID",
    "project_url":"REDCap project URL",
    "record":"REDCap record ID / PSCID",
    "redcap_event_name":"REDCap event name",
    "redcap_url":"REDCap instance URL",
    "username":"REDCap username",
    "${instrument_backend_name}_complete":"2"
}
```

REDCap LINST instrument process:
- See `tools/redcap2linst.php` usage.

#### Link(s) to related issue(s)

* #9473

#### Links to related PRs

- #8181
- #8282 (need to check)
